docs/proposals/images/machinepool-machines/inframachinepool-delete-replicas.plantuml

```diff
@@ -0,0 +1,33 @@
+@startuml inframachinepool-delete-replicas
+:InfraMachinePool delete replicas reconciliation;
+:fetch all InfraMachinePoolMachines;
+:fetch owner Machine for each InfraMachinePoolMachine;
+if (any owner Machine is missing) then
+  :wait for owner Machine to be created;
+  :requeue;
+  stop
+else
+endif
+repeat 
+  if (replica backing Machine does not exist) then (yes)
+    :delete Machine;
+    :Machine controller deletes InfraMachinePoolMachine;
+  else (no)
+  endif
+repeat while (more Machines?) is (yes)
```

Comment:
what does `more Machines?` mean?

Reply:
I meant to say "for each Machine, check if it has a replica, and if not delete it." I got the idea from [this](https://github.com/kubernetes-sigs/cluster-api/blob/5db6e22519699f4a563795ed2ac1514d4675346d/docs/book/src/images/cluster-admission-machineset-controller.plantuml#L13) but I can try to rework it.